### PR TITLE
Add size_in_bytes to enrich cache stats

### PR DESCRIPTION
**Pull request by nielsbauman**
As preparation for #106081, this PR adds the `size_in_bytes` field to the enrich cache. This field is calculated by summing the ByteReference sizes of all the search hits in the cache. It's not a perfect representation of the size of the enrich cache on the heap, but some experimentation showed that it's quite close.

### CODE REVIEW

File: docs/reference/ingest/apis/enrich/enrich-stats.asciidoc [hunk 1]
```diff
@@ -121,6 +121,10 @@ The amount of time in milliseconds spent fetching data from the cache on success
 `misses_time_in_millis`::
 (Long)
 The amount of time in milliseconds spent fetching data from the enrich index and updating the cache, on cache misses only.
+
+`size_in_bytes`::
```
Comment (masseyke) [inline docs/reference/ingest/apis/enrich/enrich-stats.asciidoc:125]: Don't we also usually document the human readable version (`size` in this case)?

Comment (masseyke) [reply docs/reference/ingest/apis/enrich/enrich-stats.asciidoc:125]: Although having said that, I see that we don't do that for `hits_time_in_millis` or `misses_time_in_millis` for this API.

**Review by masseyke — APPROVED** [review]
Looks good to me. I initially thought maybe a yaml rest test would be good, but I'm really not sure what value there would be in that (since you probably couldn't do much more than assert that the size is greater than 0).

File: docs/reference/ingest/apis/enrich/enrich-stats.asciidoc (hunk 1 above)
Comment (nielsbauman) [reply docs/reference/ingest/apis/enrich/enrich-stats.asciidoc:125]: Yeah I thought about that as well, but couldn't find an example/case where we already do that, so I decided to leave it like this.

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/enrich/20_standard_index.yml [hunk 2]
```diff
@@ -187,3 +187,61 @@ enrich documents over _bulk via an alias:
   - do:
       enrich.delete_policy:
         name: test_alias_policy
+
+---
+enrich stats REST response structure:
```
Comment (nielsbauman) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/enrich/20_standard_index.yml:192]: @masseyke @joegallo I added a YAML REST test that validates the response structure of the `_enrich/stats` API [0767cc6](https://github.com/elastic/elasticsearch/pull/110578/commits/0767cc65446ff1042c28d26bbb95959824817111). Let me know what you think (also if you think this is even valuable/worth it or not).

Comment (masseyke) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/enrich/20_standard_index.yml:192]: I'm not sure that it adds any value that enrich-stats.asciidoc doesn't already provide. But it probably doesn't hurt anything, and might be a useful building block to someone in the future. So now that you've written it, might as well leave it in.

Comment (nielsbauman) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/enrich/20_standard_index.yml:192]: Ah right, that's true. I forgot that those docs tests already provide some validation on the REST structure.

Comment (joegallo) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/enrich/20_standard_index.yml:192]: I like having the yaml test, even if there's technically some redundancy with the docs tests.

**Review by mark-vieira — APPROVED** [review]

**Comment by nielsbauman** [comment]
Merging manually because docs CI has succeeded but that isn't reflected in the status checks on the PR.

**Pull request merged**